pkg/gce-pd-csi-driver/node.go

```diff
@@ -365,7 +367,38 @@ func (ns *GCENodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGe
 }
 
 func (ns *GCENodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("NodeExpandVolume is not yet implemented"))
+	volumeID := req.GetVolumeId()
+	if len(volumeID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "ControllerExpandVolume volume ID must be provided")
+	}
+	capacityRange := req.GetCapacityRange()
+	reqBytes, err := getRequestCapacity(capacityRange)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("ControllerExpandVolume capacity range is invalid: %v", err))
+	}
+
+	volKey, err := common.VolumeIDToKey(volumeID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("ControllerExpandVolume volume ID is invalid: %v", err))
+	}
+
+	devicePath, err := ns.getDevicePath(volumeID, "")
+	if err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Error when getting device path for %s: %v", volumeID, err))
+	}
+
+	resizer := resizefs.NewResizeFs(ns.Mounter)
+	resized, err := resizer.Resize(devicePath, "")
+	if err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Error when resizing volume %s: %v", volKey.String(), err))
+
+	}
+
+	resp := &csi.NodeExpandVolumeResponse{}
+	if resized {
+		resp.CapacityBytes = reqBytes
```

Comment:
do we need to validate that the new size is actually reqBytes?

Reply:
sure, I guess we could do a `df` but is that really necessary?

Reply:
Since we don't actually pass in a size to resizefs, how do we know the size we return is actually what was requested?

Reply:
added validation